targets/unity-v2/Testing/Tests/Shared/Uunit/UUnitTestContext.cs

```diff
@@ -44,6 +44,10 @@ public UUnitTestContext(UUnitTestCase testInstance, Action<UUnitTestContext> tes
 
         public void EndTest(UUnitFinishState finishState, string resultMsg)
         {
+            if (finishState != UUnitFinishState.PENDING)
+            {
+                return;
```

Comment:
This is not sufficient.
This just hides the problem.

There are state transitions that are normal/acceptable:
PENDING->PASSED
PENDING->FAILED
PASSED->FAILED

There are state transitions that should be explicitly disallowed, and attempting them should result in failure:
FAILED->PASSED (Keep original fail message, append "You can't mark a test as finished twice")
PASSED->PASSED (Fail with a new message that says "You can't mark a test as finished twice")
FAILED->FAILED (Keep/merge both messages)